projects/packages/publicize/src/jetpack-social-settings/class-settings.php

```diff
@@ -0,0 +1,263 @@
+<?php
+/**
+ * Settings class.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize\Jetpack_Social_Settings;
+
+use Automattic\Jetpack\Modules;
+use Automattic\Jetpack\Publicize\Social_Image_Generator\Templates;
+
+/**
+ * This class is used to get and update Jetpack_Social_Settings.
+ * Currently supported features:
+ *      - Social Image Generator
+ *      - Auto Conversion
+ */
+class Settings {
+	/**
+	 * Name of the database option.
+	 *
+	 * @var string
+	 */
+	const OPTION_PREFIX = 'jetpack_social_';
+	// cSpell:ignore AUTOCONVERT
+	const AUTOCONVERT_IMAGES       = 'autoconvert_images';
+	const IMAGE_GENERATOR_SETTINGS = 'image_generator_settings';
+
+	const DEFAULT_IMAGE_GENERATOR_SETTINGS = array(
+		'enabled'  => false,
+		'template' => Templates::DEFAULT_TEMPLATE,
+	);
+
+	const DEFAULT_AUTOCONVERT_IMAGES_SETTINGS = array(
+		'enabled' => true,
+	);
+
+	/**
+	 * Migrate old options to the new settings. Previously SIG settings were stored in the
+	 * jetpack_social_image_generator_settings option. Now they are stored in the jetpack_social_settings
+	 * together with the auto conversion settings.
+	 *
+	 * TODO: Work out if this is possible on plugin upgrade
+	 *
+	 * @return void
+	 */
+	private function migrate_old_option() {
+		$auto_conversion_settings = get_option( 'jetpack_social_settings' );
+		if ( ! empty( $auto_conversion_settings['image'] ) ) {
+			update_option( self::OPTION_PREFIX . self::AUTOCONVERT_IMAGES, array( 'enabled' => $auto_conversion_settings['image'] ) );
```

Comment:
Probably need to check the logic here, but then this would be something like:

```suggestion
			update_option( self::OPTION_PREFIX . self::AUTOCONVERT_IMAGES, array( 'enabled' => ! empty( $auto_conversion_settings['image'] ) ) );
```

Reply:
https://github.com/Automattic/jetpack/pull/34113/commits/0ba898a4fa9547ec6d74da7c16b70d0257a7e0e7